lib/facade_converter.ts

```diff
@@ -88,6 +88,36 @@ export class FacadeConverter extends base.TranspilerBase {
     return handler && !handler(pa);
   }
 
+  buildImports(sourceFile: ts.SourceFile) {
+    var imports: {[type: string]: boolean} = {};
+
+    function getNodes(sourceFile: ts.SourceFile): ts.Node[] {
+      var nodes: ts.Node[] = [];
+      function allNodes(n: ts.Node){ts.forEachChild(n, n => {
+        nodes.push(n);
+        allNodes(n);
+        return false;
+      })};
+      allNodes(sourceFile);
+      return nodes;
+    }
+
+    var types = getNodes(sourceFile)
+                    .filter(n => n.kind === ts.SyntaxKind.TypeReference)
+                    .map(n => base.ident((<ts.TypeReferenceNode>n).typeName));
+
+
+    for (var iType in types) {
```

Comment:
I'm a big fan of generalizing code, but I think in this case the code would be simpler and much more readable if we'd just plain special case the `XMLHttpRequest => dart:html` case. What about calling the method `maybeEmitDartHtmlImport` and really just doing that?

In case we find we need this more often, we can still go generic later on. But I'd like to see the second and maybe third case before.